src/Monolog/Logger.php

```diff
@@ -285,9 +285,11 @@ public function addRecord($level, $message, array $context = array())
         foreach ($this->processors as $processor) {
             $record = call_user_func($processor, $record);
         }
-        while (isset($this->handlers[$handlerKey]) &&
-            false === $this->handlers[$handlerKey]->handle($record)) {
-            $handlerKey++;
+
```

Comment:
Do you mean line 258? It still returns earlier if no handler handles that message. Any suggestion on how to merge both functionalities?